spec/services/ucas_matching/send_ucas_match_emails_spec.rb

```diff
@@ -0,0 +1,62 @@
+require 'rails_helper'
+
+RSpec.describe UCASMatching::SendUCASMatchEmails, sidekiq: true do
+  let!(:ucas_match_no_action_needed) { create(:ucas_match, scheme: 'D', matching_state: 'new_match', action_taken: nil) }
+  let!(:ucas_match_action_needed) do
+    create(:ucas_match,
+           matching_state: 'new_match',
+           scheme: 'B',
+           ucas_status: :awaiting_provider_decision,
+           action_taken: nil)
+  end
+  let!(:ucas_match_to_send_remider_emails) do
+    create(:ucas_match,
+           matching_state: 'new_match',
+           scheme: 'B',
+           ucas_status: :awaiting_provider_decision,
+           action_taken: 'initial_emails_sent',
+           candidate_last_contacted_at: 8.business_days.before(Time.zone.now))
+  end
+
+  describe '#perform' do
+    before do
+      allow(ucas_match_no_action_needed).to receive(:action_needed?).and_return(false)
```

Comment:
Is this stubbing belt-and-braces given we're specifying all the properties of these objects, above? Better to have a factory?

Reply:
Definitely belt-and-braces. I think one of them was flaky and just stubbed them all and forgot to tidy this up 😬 

Following our chat today about the ucas_match factory and updating process_matching_date_spec I think I will just stub it here for now and update the factory and process_matching_date_spec in a separate PR. I identified that we may want to have up to 11 traits.